scripts/tableland/tasks.ts

```diff
@@ -497,10 +497,226 @@ task('create-manufacturer-table-schema', 'npx hardhat create-manufacturer-table-
+
+                await tablelandValidator.pollForReceiptByTransactionHash({
+                    chainId: CHAIN_ID[currentNetwork],
+                    transactionHash: (await dTx.wait())?.hash as string,
+                });
+
+                const ddTableName = await ddTableInstance.getDeviceDefinitionTableName(manufacturerId);
+
+                console.log(`Device Definition table created\nTable ID: ${ddTableId}\nTable Name: ${ddTableName}`);
+            }
+
+            ddTableId = await ddTableInstance.getDeviceDefinitionTableId(manufacturerId);
+
+            const deviceDefinitionByManufacturers = devices.filter((c) => c.make.name === make && c.type.year > 2006);
+            const ddTableName = await ddTableInstance.getDeviceDefinitionTableName(manufacturerId);
+            const tablelandDeviceDefinitionByManufacturers = await getDeviceDefinitionsByTableName(db, ddTableName);
+
+            console.log(`API Device Definition By Manufacturer [${make}] total => ${deviceDefinitionByManufacturers.length}`);
+            console.log(`Tableland Device Definition By Manufacturer [${make}] total => ${tablelandDeviceDefinitionByManufacturers.length}`);
+
+            // Insert new dd
+            let newDeviceDefinitionByManufacturers = [];
+            deviceDefinitionByManufacturers.forEach(element => {
+                const dds = tablelandDeviceDefinitionByManufacturers.filter((c) => c.id === element.name_slug);
+                if (dds != undefined && dds.length == 0){
+                    newDeviceDefinitionByManufacturers.push(element);
+                }
+            });
+
+            // Update dd when change attributes
+            let updateDeviceDefinitionByManufacturers = [];
+            deviceDefinitionByManufacturers.forEach(element => {
+                const dds = tablelandDeviceDefinitionByManufacturers.filter((c) => c.id === element.name_slug);
+                if (dds != undefined && dds.length > 0){
+
+                    if (dds[0].metadata.device_attributes) {
+                        const different = dds[0].metadata.device_attributes.filter(obj2 => {
+                            const obj1 = element.device_attributes.find(item => item.name === obj2.name);
+                            return obj1 && obj1.value !== obj2.value;
+                        });
+                          
+                        if (different.length > 0) {
+                            updateDeviceDefinitionByManufacturers.push(element);
+                        }
+                    }
+                    
+                }
+            });
+
+            // delete dd
+            let deleteDeviceDefinitionByManufacturers = [];
+            tablelandDeviceDefinitionByManufacturers.forEach(element => {
+                const dds = deviceDefinitionByManufacturers.filter((c) => c.name_slug === element.id);
+                //console.log(element.ksuid, dds[0].device_definition_id);
+                if (dds == undefined || dds == 0){
```

Comment:
should we check for length, maybe something like this:
```suggestion
                if ((dds?.length ?? 0) == 0){
```